### ISSUE TITLE
OpenQML workflow

### ISSUE THREAD

**Issue by co9olguy** [PennyLaneAI/pennylane#10]
Hi all, because of the nature of this project and our distributed team, it's become clear that we need to establish some more firm guidelines and best practices to make sure things move forward smoothly. 

The number one priority is to _keep everything moving in sync_ as much as possible. This means that everyone should, as far as possible, work on orthogonal tasks. Discussion/suggestions are ok, but please don't spend your time working on something that others have been assigned. In particular, do not write and commit new code that falls under someone else's assignments. This takes your time away from your assigned tasks, and can delay others from their tasks. 

For clarity, these are people's current assigned tasks:

- Ville: implement autograd for CV gaussian circuits (if this is done, check in about what to do next)
- Josh: design new API and refactor code to implement it
- Maria: code up examples (e.g., VQE) and verify that they work; write documentation for these
- Christian: Project Q plugin. After that, start creating fuller documentation and tests for the plugins, and work toward migrating these outside of the main library.
- Nathan: design new API w/ Josh, work on introductory sections of docs, coordinate work

Please do not make new branches! I got back from vacation yesterday and there are now 13 branches. I don't think that anyone can argue that this is a good or efficient strategy. We don't have time for code review/merging of so many branches. Please make branches only for your assigned tasks. Please remove any branches that have already been merged and are no longer needed.

If you do want to discuss/suggest things, use slack (for quick questions) or the github issues page (for more detailed discussions). Also consider using the github projects page for things that require coordination between multiple people.

Some other tasks are waiting for the final API changes. This was @josh146 's task to implement last week. Everyone is free to comment (on a github issue) about any technical issues they can find. By creating several new branches with suggestions, everything that was waiting on the finalized API is now held up.
